### ISSUE TITLE
Consider adding a /v1/restart removeContainer flag

### ISSUE THREAD

**Issue by lekkas** [balena-os/balena-supervisor#385]
It'd be nice if the `/v1/restart` endpoint accepted a `removeContainer` flag to control whether the container will be removed or not. Currently this is always the case so this endpoint can't be used by tools like `resin sync` (which require the container data to persist in-between restarts).

The current workaround for this is to make two separate calls to `/v1/apps/:appId/stop` and `/v1/apps/:appId/start`

**Comment by cywang117** [comment]
Status: currently calling restart endpoints do not remove containers, they only stop & restart them. When a service-manager 'kill' composition step is generated, the default option is to remove the container killed.

Because the original friction behind raising this issue is gone, I'm tagging this as low priority.

**Comment by 20k-ultra** [comment]
If the restart endpoint doesn't remove the container and couldn't we just close the issue ?

**Comment by cywang117** [comment]
@20k-ultra if there isn't any demand for a removeContainer flag on restart, I agree, and it doesn't look like there is (any demand, that is.)

**Comment by 20k-ultra** [comment]
@cywang117 assuming you opened this because the behaviour actually does remove the container so this feature is valid, correct ?

**Comment by cywang117** [comment]
@20k-ultra I opened this previously, when my idea for the restart action improvement was to allow users to decide whether they wanted to remove their containers or not. This user decision would be exposed as a toggle in the UI, which calls the `/vx/restart` endpoint with a request body of `{ removeContainer: boolean }`. I'd forgotten to close this issue again after the product call, where we decided to keep the actions as remove + recreate everywhere. Thanks for reminding!
